src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -168,41 +168,48 @@ void TemplateTable::shouldnotreachhere() {
 
 void TemplateTable::aconst_null() {
   transition(vtos, atos);
-  __ li_PPC(R25_tos, 0);
+  __ lui(R25_tos, 0);
 }
 
 void TemplateTable::iconst(int value) {
   transition(vtos, itos);
+  printf("iconst %d: %p\n", value, __ pc());
   assert(value >= -1 && value <= 5, "");
-  __ li_PPC(R25_tos, value);
+  if (value >= 0) {
+    __ li(R25_tos, value);
+  } else {
+    __ addi(R25_tos, R0_ZERO, -1);
+  }
 }
 
 void TemplateTable::lconst(int value) {
   transition(vtos, ltos);
-  assert(value >= -1 && value <= 5, "");
-  __ li_PPC(R25_tos, value);
+  assert(value == 0 || value == 1, "");
+  __ li(R25_tos, value);
 }
 
 void TemplateTable::fconst(int value) {
   transition(vtos, ftos);
   static float zero = 0.0;
   static float one  = 1.0;
   static float two  = 2.0;
+
   switch (value) {
     default: ShouldNotReachHere();
     case 0: {
-      int simm16_offset = __ load_const_optimized(R5_scratch1, (address*)&zero, R0, true);
-      __ lfs_PPC(F23_ftos, simm16_offset, R5_scratch1);
+      __ fmvwx(F23_ftos, R0_ZERO);
       break;
     }
     case 1: {
-      int simm16_offset = __ load_const_optimized(R5_scratch1, (address*)&one, R0, true);
-      __ lfs_PPC(F23_ftos, simm16_offset, R5_scratch1);
+      __ li(R5_scratch1, 0x3f8);
+      __ slli(R5_scratch1, R5_scratch1, 20);
```

Comment:
Could you use `lui` for this?

Reply:
BTW, after careful look, I like such approach. Const generation (1 instruction) + fmv (1) looks better than load from mem (at least 2 for addr gen + 1 load).

`fconst_0` implemented in 1 inst is beautiful :)